src/main/scala/it/pagopa/interop/purposemanagement/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -60,7 +60,17 @@ final case class PurposeApiServiceImpl(
     logger.info(operationLabel)
 
     val purpose: PersistentPurpose = PersistentPurpose.fromSeed(purposeSeed, uuidSupplier, dateTimeSupplier)
-    val result: Future[Purpose] = commander(purpose.id.toString).askWithStatus(CreatePurpose(purpose, _)).map(_.toAPI)
+    val purposeVersion: PersistentPurposeVersion =
+      PersistentPurposeVersion.fromPurposeSeed(purposeSeed, uuidSupplier, dateTimeSupplier)
+
+    val result: Future[Purpose] = for {
+      newPurpose <- commander(purpose.id.toString)
+        .askWithStatus(CreatePurpose(purpose, _))
+        .map(_.toAPI)
+      newVersion <- commander(purpose.id.toString)
+        .askWithStatus(CreatePurposeVersion(purpose.id.toString, purposeVersion, _))
+        .map(_.toAPI)
```

Comment:
these should be replaced with a single call.
The `CreatePurpose` takes a full purpose, in which you can attach the version